primitives/src/v4/mod.rs

```diff
@@ -438,6 +439,8 @@ pub struct CandidateDescriptor<H = Hash> {
 	pub relay_parent: H,
 	/// The collator's sr25519 public key.
 	pub collator: CollatorId,
+	/// The collator's `OpaquePeerId`.
+	pub peer_id: OpaquePeerId,
```

Comment:
We should not be needing this field, I think. The `CandidateDescriptor` is part of the validation. We don't need to validate (nor want to) the peer id if the collator. This is purely about connection management on the network level.

Reply:
Are checks like [these](https://github.com/paritytech/polkadot/blob/master/node/core/backing/src/lib.rs#L918-L931) then still collator based?

Reply:
We can replace it with one that checks the peer id of the connecting peer and move the check further up, to the actual reception of the message in the collator protocol